x-pack/plugins/security_solution/server/lib/telemetry/endpoint_task.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { Logger } from 'src/core/server';
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '../../../../task_manager/server';
+import { TelemetryEventsSender } from './sender';
+import { FullAgentPolicyInput } from '../../../../fleet/common/types/models/agent_policy';
+import {
+  EndpointMetricsAggregation,
+  EndpointPolicyResponseAggregation,
+  EndpointPolicyResponseDocument,
+  FleetAgentCacheItem,
+} from './types';
+
+export const TelemetryEndpointTaskConstants = {
+  TIMEOUT: '5m',
+  TYPE: 'security:endpoint-meta-telemetry',
+  INTERVAL: '24h',
+  VERSION: '1.0.0',
+};
+
+export class TelemetryEndpointTask {
+  private readonly logger: Logger;
+  private readonly sender: TelemetryEventsSender;
+
+  constructor(
+    logger: Logger,
+    taskManager: TaskManagerSetupContract,
+    sender: TelemetryEventsSender
+  ) {
+    this.logger = logger;
+    this.sender = sender;
+
+    taskManager.registerTaskDefinitions({
+      [TelemetryEndpointTaskConstants.TYPE]: {
+        title: 'Security Solution Telemetry Endpoint Metrics and Info task',
+        timeout: TelemetryEndpointTaskConstants.TIMEOUT,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          const { state } = taskInstance;
+
+          return {
+            run: async () => {
+              const lastExecutionTimestamp = moment().utc().toISOString();
```

Comment:
Should we use the [getLastTaskExecutionTimestamp](https://github.com/elastic/kibana/pull/102171/files#diff-271bdffffa10ec78f974ac82490d5f47648d16f915b6f93b7d0a9e2d343a307dR47) helper here?

Reply:
Thanks! Updated in https://github.com/elastic/kibana/pull/102171